src/thriveopedia/Thriveopedia.cs

```diff
@@ -0,0 +1,338 @@
+    {
+        var parent = GetPage(parentName);
+        var pageInTree = pageTree.CreateItem(parent != null ? allPages[parent] : null);
+
+        // Set the name of the tree item so we can reference it later
+        pageInTree.SetMeta("name", page.PageName);
+
+        // Godot doesn't appear to have a left margin for text in items, so add some manual padding
+        pageInTree.SetText(0, "  " + page.TranslatedPageName);
+
+        return pageInTree;
+    }
+
+    /// <summary>
+    ///   Selects an item in the page tree without initiating an item selected signal.
+    /// </summary>
+    /// <param name="page">The page to select</param>
+    private void SelectInTreeWithoutEvent(ThriveopediaPage page)
+    {
+        // Block signals during selection to prevent running code twice, then reattach them
+        pageTree.SetBlockSignals(true);
+        allPages[page].Select(0);
+        pageTree.SetBlockSignals(false);
+    }
+
+    private void OnPageSelectedFromPageTree()
+    {
+        var name = (string)pageTree.GetSelected().GetMeta("name");
+        ChangePage(name);
+    }
+
+    /// <summary>
+    ///    Opens an existing Thriveopedia page, optionally adding it to the page history.
+    /// </summary>
+    /// <param name="pageName">The name of the page</param>
+    /// <param name="addToHistory">Whether this page should be added to the history</param>
+    /// <param name="clearFuture">Whether this operation should clear the page future</param>
+    private void ChangePage(string pageName, bool addToHistory, bool clearFuture)
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        if (pageName == SelectedPage.PageName)
+            return;
+
+        var page = GetPage(pageName);
+
+        if (page == null)
+        {
+            GD.PrintErr($"No page with name {pageName} exists");
+            return;
+        }
+
+        if (addToHistory)
+            pageHistory.Push(page);
+
+        if (clearFuture)
+            pageFuture.Clear();
+
+        SelectedPage = page;
+    }
+
+    private void OnCollapseTreePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        pageTreeContainer.Visible = !pageTreeContainer.Visible;
+        foreach (var page in allPages.Keys)
+        {
+            page.OnNavigationPanelSizeChanged(pageTreeContainer.Visible);
+        }
+    }
+
+    private void OnHomePressed()
+    {
+        ChangePage(homePage.PageName);
+    }
+
+    private void OnBackPressed()
+    {
+        pageFuture.Push(pageHistory.Pop());
+        ChangePage(pageHistory.Peek().PageName, false, false);
+    }
+
+    private void OnForwardPressed()
+    {
+        ChangePage(pageFuture.Pop().PageName, true, false);
+    }
+
+    private void OnClosePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        Exit();
+    }
+
+    private void Exit()
+    {
+        // Reset to the home page for the next session
```

Comment:
When opened in-game it would be super useful if the history was kept between opening and closing the thriveopedia.

Reply:
I think I tried this as a fix for a bug I found with the page tree, found a better fix, and accidentally left this in. So yeah, I'll remove it.